### PR TITLE
fix test_convolution error when use cudnn.flags

### LINKED ISSUE #145473: torch.backends.cudnn.flags use error when test
### 🐛 Describe the bug

https://github.com/pytorch/pytorch/blob/main/test/nn/test_convolution.py#L3311

There is a problem with the use of cudnn.flags() here. The original purpose was to test the accuracy of allow_tf32 when it was turned on and off, but the call to cudnn.flags() causes allow_tf32 to always be True.

```python
    @onlyCUDA
    @tf32_on_and_off(0.005)
    def test_ConvTranspose2d_size_1_kernel(self, device):
        x_cpu = torch.randn(2, 3, 5, 5)
        conv_cpu = torch.nn.ConvTranspose2d(3, 3, kernel_size=1)
        y_cpu = conv_cpu(x_cpu)
        y = torch.rand_like(y_cpu)
        y_cpu.backward(y)

        with cudnn.flags(enabled=False):
            conv_cuda = torch.nn.ConvTranspose2d(3, 3, kernel_size=1).to(device)
            conv_cuda.bias.data.copy_(conv_cpu.bias.data)
            conv_cuda.weight.data.copy_(conv_cpu.weight.data)
            y_cuda = conv_cuda(x_cpu.to(device))
            y_cuda.backward(y.to(device))

        self.assertEqual(y_cpu, y_cuda, atol=1e-5, rtol=0, exact_device=False)

```

### Versions

Regardless of the environment

cc @csarofeen @ptrblck @xwang233 @eqy

### PR DESCRIPTION
**Pull request by mengph**
Fixes #145473

cc @csarofeen @ptrblck @xwang233 @eqy

### CODE REVIEW

**Review by eqy — COMMENTED** [review]
I'm not sure I understand this issue. If cuDNN is not enabled, what does setting `allow_tf32` for cuDNN only achieve?

https://github.com/pytorch/pytorch/blob/29ddf9a63ebf416cd99d277157dffacd8bc5e3bf/torch/backends/cudnn/__init__.py#L148

**Comment by mengph** [comment]
@eqy Sorry, I didn't understand the example, you are right.
